@kiva/kv-shop/src/basketCredits.ts

```diff
@@ -54,9 +54,9 @@ export interface ApplyPromoCreditData {
 export async function applyPromoCredit(
 	apollo: ApolloClient<any>,
 	options: MutationOptions<any>,
-): Promise<boolean> {
-	if (!options?.variables?.creditType && !options?.variables?.redemptionCode) {
-		return Promise.resolve(false);
+): Promise<any> {
+	if (!options?.variables?.creditType || !options?.variables?.redemptionCode) {
+		return false;
```

Comment:
Still need to return the promise here... probably should return a shop error structure. Something like...
```
{
  "errors":
    [
      {
        "message":"Missing require parameter.",
        "extensions":{"code":"upc.missing_parameter"}
      }
    ]
}
```

Reply:
done